tests/foreman/ui_airgun/test_contentcredential.py

```diff
@@ -24,712 +24,864 @@
+
+
+@tier2
+def test_positive_update_key_for_empty_product(session, module_org, gpg_content):
+    """Create gpg key with valid name and valid gpg key then associate it
+    with empty (no repos) custom product then update the key
+
+    :id: 519817c3-9b67-4859-8069-95987ebf9453
+
+    :expectedresults: gpg key is associated with product before/after
+        update
+
+    :CaseLevel: Integration
+    """
+    name = gen_string('alpha')
+    new_name = gen_string('alpha')
+    gpg_key = entities.GPGKey(
+        content=gpg_content,
+        name=name,
+        organization=module_org,
+    ).create()
+    # Creates new product and associate GPGKey with it
+    product = entities.Product(
+        gpg_key=gpg_key,
+        organization=module_org,
+    ).create()
+    with session:
+        values = session.contentcredential.read(name)
+        # Assert that GPGKey is associated with product
+        assert len(values['products']['table']) == 1
+        assert values['products']['table'][0]['Name'] == product.name
+        session.contentcredential.update(
+            name,
+            {'details.name': new_name},
+        )
+        values = session.contentcredential.read(new_name)
+        # Assert that GPGKey is still associated with product
+        assert len(values['products']['table']) == 1
+        assert values['products']['table'][0]['Name'] == product.name
+
+
+@tier2
+def test_positive_update_key_for_product_with_repo(session, module_org, gpg_content):
+    """Create gpg key with valid name and valid gpg key then associate it
+    with custom product that has one repository then update the key
+
+    :id: 02cb0601-6aa2-4589-b61e-3d3785a7e100
+
+    :expectedresults: gpg key is associated with product as well as with
+        repository after update
+
+    :CaseLevel: Integration
+    """
+    name = gen_string('alpha')
+    new_name = gen_string('alpha')
+    gpg_key = entities.GPGKey(
+        content=gpg_content,
+        name=name,
+        organization=module_org,
+    ).create()
+    # Creates new product and associate GPGKey with it
+    product = entities.Product(
+        gpg_key=gpg_key,
+        organization=module_org,
+    ).create()
+    # Creates new repository without GPGKey
+    repo = entities.Repository(
+        product=product,
+        url=FAKE_1_YUM_REPO,
+    ).create()
+    with session:
+        session.contentcredential.update(
+            name,
+            {'details.name': new_name},
+        )
+        values = session.contentcredential.read(new_name)
+        # Assert that GPGKey is still associated with product
+        assert len(values['products']['table']) == 1
+        assert values['products']['table'][0]['Name'] == product.name
+        assert len(values['repositories']['table']) == 1
+        assert values['repositories']['table'][0]['Name'] == repo.name
+
+
+@tier2
+@upgrade
+def test_positive_update_key_for_product_with_repos(session, module_org, gpg_content):
```

Comment:
Test test_positive_delete_key_for_product_with_repos satifies objective of test_positive_update_key_for_product_with_repos (duplicate test )

Rather we can change the name of test_positive_update_delete_key_for_product_with_repos and remove this test_positive_update_key_for_product_with_repos. if it make sense ?

Reply:
actually update and delete are different operations even from logical perspective, so they are not duplication